### PR TITLE
ROX-22599: Move secrets to RH GCP

### PR DESCRIPTION
**Pull request by gavin-stackrox**
Changes the deploy script to use secrets from the RH GCP project. Testing detailed in https://github.com/stackrox/automation-iac/pull/121.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -257,23 +257,23 @@ bounce-argo-pods:
 ## Secrets ##
 #############
 .PHONY: secrets-download
-secrets-download: pre-check
```
Comment (gavin-stackrox) [inline Makefile:260]: In the context of secrets this precheck only has usefulness (IMO) to check that ENVIRONMENT is set and that is handled by the secrets.sh script.

File: scripts/deploy/secrets.sh [hunk 2]
```diff
@@ -96,5 +86,4 @@ revert() {
     upload_secrets
 }
 
-check_not_empty TASK ENVIRONMENT
```
Comment (gavin-stackrox) [inline scripts/deploy/secrets.sh:99]: This is covered by `set -u` and lines 5 and 6. i.e. if those args are undefined the script fails.

Comment (tommartensen) [reply scripts/deploy/secrets.sh:99]:
```
$ .make secrets-download
./scripts/deploy/secrets.sh: line 6: $2: unbound variable
make: *** [secrets-download] Error 1
```

You're technically correct, but the error is not very useful. I would suggest to keep this error handling.

**Review by tommartensen — CHANGES_REQUESTED** [review]

File: scripts/deploy/secrets.sh (hunk 2 above)
Comment (gavin-stackrox) [reply scripts/deploy/secrets.sh:99]: > ```
> $ .make secrets-download
> ./scripts/deploy/secrets.sh: line 6: $2: unbound variable
> make: *** [secrets-download] Error 1
> ```
> 
> You're technically correct, but the error is not very useful. I would suggest to keep this error handling.

This is the case with the current handling i.e. `check_not_empty` does not run due to `set -u` https://github.com/stackrox/infra/blob/master/scripts/deploy/secrets.sh

Comment (tommartensen) [reply scripts/deploy/secrets.sh:99]: Should we remove `-u` then?

Comment (gavin-stackrox) [reply scripts/deploy/secrets.sh:99]: > Should we remove `-u` then?

`set -u` is a good thing in bash scripts. I'd like to merge this change as the script behaves as before with less code which is a net improvement IMO.

**Review by tommartensen — APPROVED** [review]

**Pull request merged**